polaris-react/src/components/Button/Button.stories.tsx

```diff
@@ -844,26 +847,45 @@ export function CopyToClipboard() {
     defaultValue: 'hello@example.com',
   });
 
+  const ref = useRef(null);
+  const isFocusedIn = useFocusIn(ref);
+  const isHovered = useHover(ref);
+  const isMouseDevice = useMediaQuery('mouse');
+  const isMouseHovered = isMouseDevice ? isHovered : true;
+
   return (
     <div style={{maxWidth: 300, paddingTop: 100}}>
       <Card>
-        <InlineStack align="space-between" gap="200">
-          <Link removeUnderline>hello@example.com</Link>
-          <Tooltip
-            dismissOnMouseOut
-            hoverDelay={500}
-            preferredPosition="above"
-            content="Copy"
-            active={status === 'copied' ? false : undefined}
-          >
-            <Button
-              variant="tertiary"
-              accessibilityLabel="Copy email address"
-              icon={status === 'copied' ? CheckIcon : ClipboardIcon}
-              onClick={copy}
-            />
-          </Tooltip>
-        </InlineStack>
+        <div ref={ref}>
+          <InlineStack align="space-between" gap="200">
+            <Link removeUnderline>hello@example.com</Link>
+            <div
+              style={{
+                opacity:
+                  isMouseHovered || isFocusedIn || status === 'copied' ? 1 : 0,
+                transition:
+                  isMouseHovered || isFocusedIn
+                    ? 'var(--p-motion-duration-100) var(--p-motion-ease) opacity'
+                    : 'none',
+              }}
+            >
+              <Tooltip
+                dismissOnMouseOut
+                hoverDelay={500}
+                preferredPosition="above"
+                content="Copy"
+                active={status === 'copied' ? false : undefined}
+              >
+                <Button
+                  variant="tertiary"
+                  accessibilityLabel="Copy email address"
+                  onClick={copy}
+                  icon={status === 'copied' ? CheckIcon : ClipboardIcon}
+                />
```

Comment:
Small suggestion to vertically center Button in the story:
```suggestion
                <InlineStack align="center">
                  <Button
                    variant="tertiary"
                    accessibilityLabel="Copy email address"
                    onClick={copy}
                    icon={status === 'copied' ? CheckIcon : ClipboardIcon}
                  />
                </InlineStack>
```

Reply:
Thanks for flagging the alignment issue! Instead of introducing an element, I added [`blockAlign="center"`](https://github.com/Shopify/polaris/pull/11802/commits/bd58cc30c9586897d6dbe762d218fe7571738665#diff-1a0163d8be285864df6922c431e39a2d79bb7832394d6f544adbc26d88ae6c7bR860) to the parent `InlineStack` and [`activatorWrapper="div"`](https://github.com/Shopify/polaris/pull/11802/commits/bd58cc30c9586897d6dbe762d218fe7571738665#diff-1a0163d8be285864df6922c431e39a2d79bb7832394d6f544adbc26d88ae6c7bR878) to the `Tooltip`:

Before:
![Screenshot 2024-04-01 at 11 39 59 AM](https://github.com/Shopify/polaris/assets/32409546/c83752c2-a97f-4e46-bc73-5c632452c522)

After:
![Screenshot 2024-04-01 at 11 40 07 AM](https://github.com/Shopify/polaris/assets/32409546/b45f4d11-5ade-4898-b21d-12576b4673c4)

Before:
![Screenshot 2024-04-01 at 11 40 49 AM](https://github.com/Shopify/polaris/assets/32409546/955df560-e2f7-486c-81c6-ac714b28d80d)

After:
![Screenshot 2024-04-01 at 11 41 35 AM](https://github.com/Shopify/polaris/assets/32409546/aa793b52-9321-4f50-ad84-32a189a16a4b)
![Screenshot 2024-04-01 at 11 41 47 AM](https://github.com/Shopify/polaris/assets/32409546/335a13d9-74e8-4dbc-a0ff-cf8d36e4f563)